### PR TITLE
Field data returns object

### PR DESCRIPTION
**Pull request by prmukherj**
FieldData modules like: "get_scalar_field_data", "get_surface_data", "get_vector_field_data" and "get_pathlines_field_data" to return objects instead of dictionaries.

Example Use Case:
```
>>> abs_press_data = field_data.get_scalar_field_data(field_name="absolute-pressure", surface_name="cold-inlet")
>>> abs_press_data.size
>>> abs_press_data[120].scalar_data

>>> velocity_vector_data = field_data.get_vector_field_data(field_name="velocity", surface_name="cold-inlet")
>>> velocity_vector_data.size
>>> velocity_vector_data.scale

>>> vertices_data = field_data.get_surface_data(data_type=SurfaceDataType.Vertices, surface_name="cold-inlet")
>>> vertices_data[5].x

>>> path_lines_data = field_data.get_pathlines_field_data(field_name="velocity", surface_name="cold-inlet")
>>> path_lines_data["lines"][100].node_count
>>> path_lines_data["lines"][100].node_data
```

### CODE REVIEW

**Comment by seanpearsonuk** [comment]
@prmukherj, Please could you provide a PR description. Thanks.

File: src/ansys/fluent/core/services/field_data.py [hunk 1]
```diff
@@ -838,6 +838,134 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+class ScalarFieldData(_FieldDataBase):
+    class _ScalarData:
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        self._data = []
+        for _data in data:
+            self._data.append(ScalarFieldData._ScalarData(_data))
+        super().__init__(i_d, self._data)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:871]: @prmukherj we can remove all lives except for the `super` line and we can pass a list comprehension as the second (data) argument.

Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Thank you @seanpearsonuk. This is addressed.

File: src/ansys/fluent/core/services/field_data.py [hunk 2]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        return self._data[item]
+
+
+class ScalarFieldData(_FieldDataBase):
+    class _ScalarData:
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData._ScalarData(_data) for _data in data])
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:868]: cleaner

File: src/ansys/fluent/core/services/field_data.py [hunk 3]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+
+class Vector:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:871]: @prmukherj , In a follow-up task - when you have time - you could add docstrings.

File: src/ansys/fluent/core/services/field_data.py [hunk 4]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class _FieldDataBase:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:841]: @prmukherj , This is actually public - it contributes to the public interfaces of the derived class - so we should give it a public name. I would suggest `BaseFieldData` because it reads better.

File: src/ansys/fluent/core/services/field_data.py [hunk 5]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        return self._z
+
+
+class VectorFieldData(_FieldDataBase):
+    class _VectorData(Vector):
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data, scale):
+        data.shape = data.size // 3, 3
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:896]: @prmukherj should we raise an exception `if data.size % 3` ?

File: src/ansys/fluent/core/services/field_data.py [hunk 6]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        data.shape = data.size // 3, 3
+        super().__init__(i_d, [(Vertices._Vertex(x, y, z)) for x, y, z in data])
+
+
+class FacesCentroid(_FieldDataBase):
+    class _Centroid(Vector):
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:918]: @prmukherj I wonder if we should un-nest `_Centroid` and `_Faces`, and gives them public names. This might clarify the documentation somewhat.

File: src/ansys/fluent/core/services/field_data.py (hunk 3 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py:871]: I'll add docstrings as part of this PR only.

File: src/ansys/fluent/core/services/field_data.py (hunk 4 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py:841]: ok, sure. Thank you

File: src/ansys/fluent/core/services/field_data.py [hunk 7]
```diff
@@ -838,6 +838,121 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    def __init__(self, i_d, data):
+        faces_data = []
+        i = 0
+
+        while i < len(data):
+            faces_data.append(
+                FacesConnectivity._Faces(data[i], data[i + 1 : i + 1 + data[i]])
+            )
+            i = i + 1 + data[i]
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:941]:
```suggestion
            end = i + 1 + data[i]
            faces_data.append(
                FacesConnectivity._Faces(data[i], data[i + 1 : end])
            )
            i = end
```

File: src/ansys/fluent/core/services/field_data.py (hunk 6 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py:918]: ok

File: src/ansys/fluent/core/services/field_data.py (hunk 5 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/services/field_data.py:896]: Perhaps also some type hints would help readability?

File: src/ansys/fluent/core/services/field_data.py [hunk 8]
```diff
@@ -1076,13 +1242,22 @@ def get_vector_field_data(
...
-        return {
-            surface_id: (
-                vector_field_data[surface_id][field_name],
-                vector_field_data[surface_id]["vector-scale"][0],
+
+        if surface_name:
+            return VectorFieldData(
+                surface_ids[0],
+                vector_field_data[surface_ids[0]][field_name],
+                vector_field_data[surface_ids[0]]["vector-scale"][0],
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/field_data.py:1250]: The "vector-scale" string originates in the FieldData gRPC code. Would it be correct to say that we need to mention this string in the proto file, so that the service can be used effectively?

File: src/ansys/fluent/core/services/field_data.py [hunk 9]
```diff
@@ -860,7 +862,11 @@ def __getitem__(self, item):
 
 
 class ScalarFieldData(BaseFieldData):
+    """Get scalar field data on a surface."""
```
Comment (ajain-work) [inline src/ansys/fluent/core/services/field_data.py:865]: Store?

File: src/ansys/fluent/core/services/field_data.py [hunk 10]
```diff
@@ -923,7 +943,11 @@ def __init__(self, i_d, data):
 
 
 class FacesConnectivity(BaseFieldData):
+    """Get surface data (faces connectivity)."""
+
     class Faces:
+        """Stores and provides the data as a vector of Faces."""
```
Comment (ajain-work) [inline src/ansys/fluent/core/services/field_data.py:949]: Not vector

File: src/ansys/fluent/core/services/field_data.py (hunk 3 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py:877]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 4 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 6 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 9 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done, Thank you

File: src/ansys/fluent/core/services/field_data.py (hunk 10 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 7 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done.

File: src/ansys/fluent/core/services/field_data.py (hunk 5 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py [hunk 11]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
     return fields_data
 
 
+class BaseFieldData:
+    """Contain common properties required by all field data types."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:842]:
```suggestion
    """Contains common properties required by all field data types."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 12]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+    @property
+    def size(self):
+        return len(self._data)
+
+    def __getitem__(self, item):
+        return self._data[item]
+
+
+class ScalarFieldData(BaseFieldData):
+    """Container for scalar field data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:865]:
```suggestion
    """Provides a container for scalar field data."""
```
We want class descriptions to start with a verb ending in "s" or "es" so that an autosummary table is formatted nicely (and consistently).

File: src/ansys/fluent/core/services/field_data.py [hunk 13]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class ScalarData:
+        """Store and provide the data as a scalar."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:868]:
```suggestion
        """Stores and provides the data as scalar data."""
```
Because "scalar" is a noun, I added the noun "data" after it. However, this could possible be just "Stores and provides scalar data." Your call!

File: src/ansys/fluent/core/services/field_data.py [hunk 14]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData.ScalarData(_data) for _data in data])
+
+
+class Vector:
+    """Stores the data in form of a vector (x, y, z)"""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:878]:
```suggestion
    """Stores the data in form of a vector ``(x, y, z)``."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 15]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+    @property
+    def z(self) -> float:
+        return self._z
+
+
+def _resolve_into_array_of_vectors(data):
+    if data.size % 3:
+        raise ValueError(
+            "Data set has to be resolved as set of vectors."
+            "The length of data set should always be in multiples of 3."
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:902]:
```suggestion
            "Dataset must be resolved as a set of vectors."
            "The length of the dataset should always be in multiples of 3."
```
Google developer documentation style guide says dataset is one word.

File: src/ansys/fluent/core/services/field_data.py [hunk 16]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        )
+    data.shape = data.size // 3, 3
+
+
+class VectorFieldData(BaseFieldData):
+    """Container for vector field data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:908]:
```suggestion
    """Provides a container for vector field data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 17]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class VectorData(Vector):
+        """Store and provide the data as vector."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:911]:
```suggestion
        """Stores and provides the data as a vector."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 18]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        self._scale = scale
+        super().__init__(i_d, [VectorFieldData.VectorData(x, y, z) for x, y, z in data])
+
+    @property
+    def scale(self) -> float:
+        return self._scale
+
+
+class Vertices(BaseFieldData):
+    """Container for Vertex data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:927]:
```suggestion
    """Provides a container for vertex data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 19]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class Vertex(Vector):
+        """Store and provide the data as a vector of Vertex."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:930]:
```suggestion
        """Stores and provides the data as a vector of a vertex."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 20]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        _resolve_into_array_of_vectors(data)
+        super().__init__(i_d, [(Vertices.Vertex(x, y, z)) for x, y, z in data])
+
+
+class FacesCentroid(BaseFieldData):
+    """Container for Faces Centroid data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:941]:
```suggestion
    """Provides the container for faces centroid data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 21]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class Centroid(Vector):
+        """Store and provide the data as a vector of Faces Centroid."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:944]:
```suggestion
        """Stores and provides the data as a vector of the faces centroid."""
```

File: src/ansys/fluent/core/services/field_data.py (hunk 20 above)
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:941]:
```suggestion
    """Provides the container for faces centroid data."""
```
I'm not sure here and in subsequent descriptions if faces should be possessive or if its plural. It seems like it should be face's centroid data., indicating the possessive form (where the centroid data belongs to a face.

File: src/ansys/fluent/core/services/field_data.py [hunk 22]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        def __init__(self, x, y, z):
+            super().__init__(x, y, z)
+
+    def __init__(self, i_d, data):
+        _resolve_into_array_of_vectors(data)
+        super().__init__(i_d, [(FacesCentroid.Centroid(x, y, z)) for x, y, z in data])
+
+
+class FacesConnectivity(BaseFieldData):
+    """Container for Faces Connectivity data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:955]:
```suggestion
    """Provides the container for face connectivity data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 23]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class Faces:
+        """Store and provide the data as an array of Faces Connectivity."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:958]:
```suggestion
        """Stores and provides the data as an array of the face's connectivity."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 24]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+        while i < len(data):
+            end = i + 1 + data[i]
+            faces_data.append(FacesConnectivity.Faces(data[i], data[i + 1 : end]))
+            i = end
+
+        super().__init__(i_d, faces_data)
+
+
+class FacesNormal(BaseFieldData):
+    """Container for Faces Normal data."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:977]:
```suggestion
    """Provides the container for the face's Normal data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 25]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+    class Normal(Vector):
+        """Store and provide the data as a vector of Faces Normal."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:980]:
```suggestion
        """Stores and provides the face's normal data as a vector."""
```

File: src/ansys/fluent/core/services/field_data.py (hunk 16 above)
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:908]:
```suggestion
    """Provides a container for vector field data."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 26]
```diff
@@ -942,8 +1091,9 @@ def get_scalar_field_data(
 
         Returns
         -------
-        Dict[int, np.array]
-            Dictionary containing a map of surface IDs to the scalar field.
+        Union[ScalarFieldData, Dict[int, ScalarFieldData]]
+            ScalarFieldData if surface name is provided as input.
+            Dictionary containing a map of surface IDs to ScalarFieldData if surface_ids are provided as input.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1096]:
```suggestion
            If a surface name is provided as input, scalar field data is returned. If surface
            IDs are provided as input, a dictionary containing a map of surface IDs to scalar
            field data.
```
The data type shows the object names, allowing the description to be in plain English.

File: src/ansys/fluent/core/services/field_data.py [hunk 27]
```diff
@@ -998,8 +1159,12 @@ def get_surface_data(
...
         Returns
         -------
-        Dict[int, np.array]
-            Dictionary containing a map of surface IDs to surface data.
+        Union[Vertices, FacesConnectivity, FacesNormal, FacesCentroid,
+        Dict[int, Union[Vertices, FacesConnectivity, FacesNormal, FacesCentroid]]]
+            Either of Vertices, FacesConnectivity, FacesNormal, or FacesCentroid
+            if surface name is provided as input.
+            Dictionary containing a map of surface IDs to Either of Vertices, FacesConnectivity,
+            FacesNormal, or FacesCentroid if surface_ids are provided as input.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1167]:
```suggestion
             If a surface name is provided as input, face vertices, connectivity, normal, or centroid data.
             If surface IDs are provided as input, a dictionary containing a map of surface IDs to face
             vertices, connectivity, normal, or centroid data.
```

File: src/ansys/fluent/core/services/field_data.py [hunk 28]
```diff
@@ -1053,8 +1258,10 @@ def get_vector_field_data(
 
         Returns
         -------
-        Dict[int, Tuple[np.array, float]]
-            Dictionary containing a map of surface IDs to a tuple of vector field and vector scale.
+        Union[VectorFieldData, Dict[int, VectorFieldData]]
+            VectorFieldData if surface name is provided as input.
+            Dictionary containing a map of surface IDs to VectorFieldData
+            if surface_ids are provided as input.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1264]:
```suggestion
            If a surface name is provided as input, vector field data.
            If surface IDs are provided as input, a dictionary containing a map of
            surface IDs to vector field data.
```

File: src/ansys/fluent/core/services/field_data.py [hunk 29]
```diff
@@ -1101,7 +1317,7 @@ def get_pathlines_field_data(
         coarsen: Optional[int] = 1,
         velocity_domain: Optional[str] = "all-phases",
         zones: Optional[list] = [],
-    ) -> Dict[int, Dict[str, np.array]]:
+    ) -> Dict:
         """Get pathlines field data on a surface.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1321]: Again, should this be possessive: Get the pathline's field data on a surface.

**Review by ajain-work — APPROVED** [review]

File: tests/test_field_data.py [hunk 30]
```diff
@@ -6,6 +6,7 @@
 from ansys.fluent.core.services.field_data import (
     ScalarFieldNameError,
     ScalarFieldUnavailable,
+    SurfaceDataType,
     SurfaceNameError,
 )
```
Comment (ajain-work) [inline tests/test_field_data.py:12]: @prmukherj Doc must be updated.

Comment (prmukherj) [reply tests/test_field_data.py:12]: Updated the doc. @PipKat could you please have a look. Thank you.

File: doc/source/api/solver/fielddata.rst [hunk 31]
```diff
@@ -45,65 +47,99 @@ the ``get_surface_data`` method and specifying ``Vertices`` for ``data_type``.
...
-  
-  >>> field_data.get_surface_data(surface_name="inlet", data_type=SurfaceDataType.FacesCentroid)
-  {4: array([-0.34724122,  0.        , -0.0442204 , -0.34724477, ...,
-         -0.04050309, -0.34645557,  0.        , -0.04421487, -0.34646705], dtype=float32)}
-       
-Response contains a dictionary of surface IDs to numpy array of the requested field. 
+
+If a surface name is provided as input response contains, face vertices, connectivity, normal, or centroid data.
+If surface IDs are provided as input response contains, a dictionary containing a map of surface IDs to face
+vertices, connectivity, normal, or centroid data..
```
Comment (PipKat) [inline doc/source/api/solver/fielddata.rst:91]:
```suggestion
If a surface name is provided as an input, the response contains face vertices, connectivity data, and normal or centroid data.
If surface IDs are provided as the input, the response is a dictionary containing a map of surface IDs to face
vertices, connectivity data, and normal or centroid data.
```
I'm assuming that the changes I made are what you mean. Please revise if incorrect!

File: doc/source/api/solver/fielddata.rst [hunk 32]
```diff
@@ -45,65 +47,99 @@ the ``get_surface_data`` method and specifying ``Vertices`` for ``data_type``.
...
 
 Get scalar field data
 ~~~~~~~~~~~~~~~~~~~~~
-You can call the ``get_scalar_field_data`` method to get scalar field data, such as temperature:
+You can call the ``get_scalar_field_data`` method to get scalar field data, such as absolute-pressure:
```
Comment (PipKat) [inline doc/source/api/solver/fielddata.rst:95]:
```suggestion
You can call the ``get_scalar_field_data`` method to get scalar field data, such as absolute pressure:
```

File: src/ansys/fluent/core/services/field_data.py [hunk 33]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+
+        def __init__(self, data):
+            self.scalar_data = data
+
+    def __init__(self, i_d, data):
+        super().__init__(i_d, [ScalarFieldData.ScalarData(_data) for _data in data])
+
+
+class Vector:
+    """Stores the data in form of a vector ``(x, y, z)``."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:878]:
```suggestion
    """Stores the data as a vector ``(x, y, z)``."""
```

File: src/ansys/fluent/core/services/field_data.py [hunk 34]
```diff
@@ -838,6 +838,155 @@ def _extract_field(field_datatype, field_size, chunk_iterator):
...
+            i = end
+
+        super().__init__(i_d, faces_data)
+
+
+class FacesNormal(BaseFieldData):
+    """Provides the container for the face normal data."""
+
+    class Normal(Vector):
+        """Stores and provides the face's normal data as a vector."""
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:980]: I'm not sure about my earlier suggestion for ussing the possessive here. Based on the description for the parent class, should we be saving "face normal data" in the description for the ``Normal`` class--or should we be using the possessive  "face's normal data`` in the parent ``FacesNormal`` class? It seems the terminology should match.

File: src/ansys/fluent/core/services/field_data.py [hunk 35]
```diff
@@ -998,8 +1160,11 @@ def get_surface_data(
 
         Returns
         -------
-        Dict[int, np.array]
-            Dictionary containing a map of surface IDs to surface data.
+        Union[Vertices, FacesConnectivity, FacesNormal, FacesCentroid,
+        Dict[int, Union[Vertices, FacesConnectivity, FacesNormal, FacesCentroid]]]
+             If a surface name is provided as input, face vertices, connectivity, normal, or centroid data.
+             If surface IDs are provided as input, a dictionary containing a map of surface IDs to face
+             vertices, connectivity, normal, or centroid data.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1167]:
```suggestion
             If a surface name is provided as input, face vertices, connectivity data, and normal or centroid data are returned.
             If surface IDs are provided as input, a dictionary containing a map of surface IDs to face
             vertices, connectivity data, and normal, or centroid data is returned.
```
Based on earlier recommendations. Given the length of the options here, it seemed to note that these are two different return values.

File: src/ansys/fluent/core/services/field_data.py [hunk 36]
```diff
@@ -1053,8 +1258,10 @@ def get_vector_field_data(
 
         Returns
         -------
-        Dict[int, Tuple[np.array, float]]
-            Dictionary containing a map of surface IDs to a tuple of vector field and vector scale.
+        Union[VectorFieldData, Dict[int, VectorFieldData]]
+            If a surface name is provided as input, vector field data.
+            If surface IDs are provided as input, a dictionary containing a map of
+            surface IDs to vector field data.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1264]:
```suggestion
        Union[VectorFieldData, Dict[int, VectorFieldData]]
            If a surface name is provided as input, vector field data is returned.
            If surface IDs are provided as input, a dictionary containing a map of
            surface IDs to vector field data is returned.
```
Again, I think adding "is returned" to the two options helps with understanding.

File: src/ansys/fluent/core/services/field_data.py [hunk 37]
```diff
@@ -1138,7 +1354,7 @@ def get_pathlines_field_data(
 
         Returns
         -------
-        Dict[int, Dict[str, np.array]]
+        Dict
             Dictionary containing a map of surface IDs to the pathline data
```
Comment (PipKat) [inline src/ansys/fluent/core/services/field_data.py:1358]:
```suggestion
            Dictionary containing a map of surface IDs to the pathline data.
```

File: tests/test_field_data.py (hunk 30 above)
Comment (PipKat) [reply tests/test_field_data.py:12]: @prmukherj Just a few minor edits--and questions to ensure that any suggestions have the meaning that you intended!

**Review by mkundu1 — APPROVED** [review]

File: src/ansys/fluent/core/services/field_data.py (hunk 34 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py:980]: Actually faces means a number of faces (surfaces). That's why the terminology FacesNormal and that's why face's normal does not make much sense in this case. You have any suggestions on what can be done to maintain consistency? @PipKat

**Review by hpohekar — APPROVED** [review]

File: tests/test_field_data.py (hunk 30 above)
Comment (prmukherj) [reply tests/test_field_data.py:12]: Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 34 above)
Comment (prmukherj) [reply src/ansys/fluent/core/services/field_data.py]: "face normal data" makes sense here.

File: doc/source/api/solver/fielddata.rst [hunk 38]
```diff
@@ -45,65 +47,99 @@ the ``get_surface_data`` method and specifying ``Vertices`` for ``data_type``.
...
+.. code-block:: python
+
+  >>> faces_normal_data = field_data.get_surface_data(
+  >>>     data_type=SurfaceDataType.FacesNormal, surface_name="cold-inlet"
+  >>> )
 
-You can call the same method to get the corresponding surface face connectivity, normals, and centroids
-by specifying ``FacesConnectivity``, ``FacesNormal`` and ``FacesCentroid`` respectively for ``data_type``.
+  >>> faces_normal_data = field_data.get_surface_data(
+  >>>     data_type=SurfaceDataType.FacesCentroid, surface_name="cold-inlet"
```
Comment (ajain-work) [inline doc/source/api/solver/fielddata.rst:72]: centroid

File: doc/source/api/solver/fielddata.rst [hunk 39]
```diff
@@ -45,65 +47,99 @@ the ``get_surface_data`` method and specifying ``Vertices`` for ``data_type``.
...
 .. code-block:: python
 
-  >>> field_data.get_surface_data(surface_name="inlet", data_type=SurfaceDataType.FacesConnectivity)
-  {4: array([  4,   3,   2, ..., 379, 382, 388])}
+  >>> faces_connectivity_data = field_data.get_surface_data(
+  >>>     data_type=SurfaceDataType.FacesConnectivity, surface_name="cold-inlet"
+  >>> )
+  >>> faces_connectivity_data[5].node_count
+  4
+  >>> faces_connectivity_data[5].node_data
```
Comment (ajain-work) [inline doc/source/api/solver/fielddata.rst:85]: node_indices

File: doc/source/api/solver/fielddata.rst (hunk 38 above)
Comment (prmukherj) [reply doc/source/api/solver/fielddata.rst:72]: Done.

File: doc/source/api/solver/fielddata.rst (hunk 39 above)
Comment (prmukherj) [reply doc/source/api/solver/fielddata.rst]: Done, Thank you.

File: src/ansys/fluent/core/services/field_data.py (hunk 8 above)
Comment (ajain-work) [reply src/ansys/fluent/core/services/field_data.py:1291]: Response only contains payload. We can discuss how to extend.

File: src/ansys/fluent/core/services/field_data.py (hunk 34 above)
Comment (PipKat) [reply src/ansys/fluent/core/services/field_data.py]: @prmukherj I'm fine with face normal data. However, the use of pathlines (pluarl) is fine. Let me find the line.
